java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -818,23 +822,47 @@ public ColumnView[] splitAsViews(int... indices) {
         nativeHandles[i] = 0;
       }
     } catch (Throwable t) {
-      cleanupColumnViews(nativeHandles, columnViews);
+      try {
+        cleanupColumnViews(nativeHandles, columnViews);
+      } catch (Throwable s) {
+        t.addSuppressed(s);
+      }
       throw t;
     }
     return columnViews;
   }
 
-  static void cleanupColumnViews(long[] nativeHandles, ColumnView[] columnViews) {
-    for (ColumnView columnView: columnViews) {
+  static void cleanupColumnViews(long[] nativeHandles, ColumnView[] columnViews) throws Throwable {
```

Comment:
This method is only called in practice when cleaning up from an exception, so rather than adding the complexity of first-vs-subsequent exceptions, this could simply always take a Throwable parameter and simply add a suppressed exception for every error caught during cleanup.

Reply:
Good point. My assumption was that we were calling it from outside of the catch block.